src/runtimes/node/bundlers/nft/es_modules.ts

```diff
@@ -49,4 +30,74 @@ const patchESMPackage = async (path: string, fsCache: FsCache) => {
   return JSON.stringify(patchedPackageJson)
 }
 
-export { getPatchedESMPackages }
+const shouldTranspile = (path: string, cache: Map<string, boolean>, reasons: NodeFileTraceReasons): boolean => {
+  const reason = reasons.get(path)
+
+  // This isn't an expected case, but if the path doesn't exist in `reasons` we
+  // don't transpile it.
+  if (reason === undefined) {
+    cache.set(path, false)
+
+    return false
+  }
+
+  const { parents } = reason
+
+  // If the path doesn't have any parents, it's safe to transpile.
+  if (parents.size === 0) {
+    cache.set(path, true)
+
+    return true
+  }
+
+  // The path should be transpiled if every parent will also be transpiled, or
+  // if there is no parent.
+  const shouldTranspilePath = [...parents].every((parentPath) => shouldTranspile(parentPath, cache, reasons))
```

Comment:
now with 1d322f3f1866b8be4a25e78fcfbb52a5f78935b5, we're writing into the cache, but never reading from it

Reply:
OOF! I need coffee. Okay, https://github.com/netlify/zip-it-and-ship-it/pull/790/commits/c66640d8fcabeb44d27ada71afe55eab37e0ff8f should do it?

Reply:
yep!